### PR TITLE
Trace2: use system config for default trace2 settings

### PR DESCRIPTION
**Pull request by jeffhostetler**
Allow trace2 settings to be loaded from the system config.
This allows administrators to set system-wide default trace2 settings.
The existing GIT_TR2* environment variables can be used to override the system defaults.

For example, this allows us to enable event tracing during VFS4G installation, rather
than forcing a change in the user's environment variables and asking them to open
new shells.  Setting values in the system config requires the use of an Administrator
Prompt.

    $ git config --system trace2.eventTarget _pathname_

This is being concurrently submitted upstream, so some re-work is possible.
https://github.com/gitgitgadget/git/pull/169

### CODE REVIEW

File: trace2/tr2_dst.c [hunk 1]
```diff
@@ -57,37 +56,103 @@ static int tr2_dst_try_path(struct tr2_dst *dst, const char *tgt_value)
...
+		close(fd);
+		return e;
+	}
+
+	*out_fd = fd;
+	return 0;
+}
+
+#define TR2_DST_UDS_TRY_STREAM (1 << 0)
+#define TR2_DST_UDS_TRY_DGRAM  (1 << 1)
```
Comment (dscho) [inline trace2/tr2_dst.c]: I do not quite understand how these changes relate to the commit message. Were these changes intended for another commit?

File: common-main.c [hunk 2]
```diff
@@ -27,6 +27,8 @@ int main(int argc, const char **argv)
 {
 	int result;
 
+	trace2_initialize_clock();
+
```
[the remark this answers is not in the document]
Comment (jeffhostetler) [reply common-main.c]: Yeah, I handle that down in the trace2/tr2_tls layer.  I want common-main.c to have the call for the Linux/Mac crowd.
Putting one here lets us also capture the mingw startup overhead on Windows.  And without any ifdef's in the common
code.

File: trace2/tr2_dst.c (hunk 1 above)
Comment (jeffhostetler) [reply trace2/tr2_dst.c]: Yeah, I need to re-do the port of the trace2 changes from upstream.  After I did this port, I realized that there must be another round of upstream changes that need to be pulled over before this system-config set is applied.

I'm about to push another version of the system-config changes upstream, so I'll revisit this shortly.

**Comment by jeffhostetler** [comment]
I'm aborting this PR.  I will re-port these (and a few other) upstream changes and target 2.21.
